code/src/containers/forms/new-product-form/NewProductForm.tsx

```diff
@@ -188,62 +191,67 @@ const NewProductForm = () => {
               variant="dark"
               labelTranslationKey="productForm.inputLabel.status"
               labelClassName="product-form__visibility-checkbox-label"
+              data-testid="new-product-status-checkbox"
               {...register("status")}
             />
           </AppBox>
         </AppBox>
         <AppBox className="product-form__main-info-section">
-          {supportedLocales.map((locale, index) => (
-            <AppBox className="product-form__container" key={locale}>
-              <AppBox className="product-form__header">
-                <AppTypography
-                  component="h1"
-                  variant="h3"
-                  translationKey={`productForm.section.mainInformation.title.${locale}`}
-                  className="product-form__header-title"
-                />
-              </AppBox>
-              <AppBox className="product-form__body">
-                <AppInput
-                  className="product-form__text-input"
-                  fullWidth
-                  labelTranslationKey="productForm.inputLabel.name"
-                  error={Boolean(
-                    errors.productTranslations &&
-                      errors.productTranslations[index]?.name
-                  )}
-                  helperText={
-                    errors.productTranslations &&
-                    errors.productTranslations[index]?.name
-                      ? errors.productTranslations[index]?.name.message
-                      : undefined
-                  }
-                  {...register(`productTranslations.${index}.name`)}
-                />
-                <AppInput
-                  className="product-form__text-input"
-                  fullWidth
-                  multiline
-                  labelTranslationKey="productForm.inputLabel.description"
-                  inputProps={{
-                    className: "product-form__description-input"
-                  }}
-                  rows={5}
-                  error={Boolean(
-                    errors.productTranslations &&
-                      errors.productTranslations[index]?.description
-                  )}
-                  helperText={
-                    errors.productTranslations &&
-                    errors.productTranslations[index]?.description
-                      ? errors.productTranslations[index]?.description.message
-                      : undefined
-                  }
-                  {...register(`productTranslations.${index}.description`)}
-                />
+          {supportedLocales.map((locale, index) => {
+            const isNameError = Boolean(
+              errors.productTranslations?.[index]?.name
+            );
```

Comment:
move it above the return statement

Reply:
Thanks for suggestion. Moved it and one more complicated part above